BitList.cs

```diff
@@ -0,0 +1,1553 @@
+    /// </summary>
+    /// <exception cref="BooleanAlgebraException" />
+    public void Not()
+    {
+        if (IsEmpty)
+            throw new BooleanAlgebraException();
+        else
+            bits.Not();
+    }
+
+    /// <summary>
+    /// AND boolean operation of this instance.
+    /// </summary>
+    /// <exception cref="BooleanAlgebraException" />
+    public bool And()
+    {
+        if (IsEmpty)
+            throw new BooleanAlgebraException();
+        else
+        {
+            bool result = bits[0];
+            for (int i = 1; i < bits.Length; i++)
+                result &= bits[i];
+            return result;
+        }
+    }
+
+    /// <summary>
+    /// AND boolean operation with another list of bit.
+    /// </summary>
+    /// <exception cref="BooleanAlgebraException" />
+    public void And(BitList bits)
+    {
+        if (IsEmpty)
+            throw new BooleanAlgebraException();
+        else
+            this.bits.And(bits.bits);
+    }
+
+    /// <summary>
+    /// OR boolean operation in this instance.
+    /// </summary>
+    /// <exception cref="BooleanAlgebraException" />
+    public bool Or()
+    {
+        if (IsEmpty)
+            throw new BooleanAlgebraException();
+        else
+        {
+            bool result = bits[0];
+            for (int i = 1; i < bits.Length; i++)
+                result |= bits[i];
+            return result;
+        }
+    }
+
+    /// <summary>
+    /// OR boolean operation with another list of bit.
+    /// </summary>
+    /// <exception cref="BooleanAlgebraException" />
+    public void Or(BitList bits)
+    {
+        if (IsEmpty)
+            throw new BooleanAlgebraException();
+        else
+            this.bits.Or(bits.bits);
+    }
+
+    /// <summary>
+    /// XOR boolean operation in this instance.
+    /// </summary>
+    /// <exception cref="BooleanAlgebraException" />
+    public bool Xor()
+    {
+        if (IsEmpty)
+            throw new BooleanAlgebraException();
+        else
+        {
+            bool result = bits[0];
+            for (int i = 1; i < bits.Length; i++)
+                result ^= bits[i];
+            return result;
+        }
+    }
+
+    /// <summary>
+    /// XOR boolean operation with another list of bit.
+    /// </summary>
+    /// <exception cref="BooleanAlgebraException" />
+    public void Xor(BitList bits)
+    {
+        if (IsEmpty)
+            throw new BooleanAlgebraException();
+        else
+            this.bits.Xor(bits.bits);
+    }
+
+    /// <summary>
+    /// Add a list of bit to the last position of this instance.
+    /// </summary>
+    /// <param name="bits">A list of bit.</param>
+    public void Add(BitList bits)
+    {
+        var newBitList = IsEmpty ? bits : this + bits;
+        this.bits = newBitList.bits;
+    }
+
+    /// <summary>
+    /// Add a bit to the last position of this instance.
+    /// </summary>
+    /// <param name="bit">A bit.</param>
+    public void Add(bool bit)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename parameter 'bit' to 'item' to match the interface declaration.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5545753)